src/scheduler/limits.c

```diff
@@ -2803,11 +2804,15 @@ check_max_group_res_soft(resource_resv *rr, counts *cts_list, void *limitctx, in
 		} else if (max_gengroup_res_soft < used) {
 			if (rescts != NULL)
 				rescts->soft_limit_preempt_bit = preempt_bit;
-			return (preempt_bit);
+			rc = preempt_bit;
+		} else {
```

Comment:
just say "else if (rescts != NULL) "?

Reply:
I purposefully did this because I wanted to have this else for the case where generic limit is under the usage. Within that case when resource counts are not null I rest the preempt_bit.
What you are suggesting is correct but for better readability I kept it the way it is, If you feel strongly about it, I'll change it.